### PR TITLE
support table filter

### PR DESCRIPTION
**Pull request by zhangjinpeng1987**

### CODE REVIEW

**Comment by siddontang** [comment]
PTAL @huachaohuang

File: librocksdb_sys/crocksdb/c.cc [hunk 1]
```diff
@@ -2758,6 +2758,32 @@ void crocksdb_readoptions_set_ignore_range_deletions(
   opt->rep.ignore_range_deletions = v;
 }
 
+struct TableFilter {
+  TableFilter(void* ctx,
+    int(*table_filter)(void*, const crocksdb_table_properties_t*),
```
Comment (huachaohuang) [inline librocksdb_sys/crocksdb/c.cc]: Better add space between return type and function name.

File: src/table_filter.rs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    // given table based on the table's properties. The callback is passed the
+    // properties of each table during iteration. If the callback returns false,
+    // the table will not be scanned. This option only affects Iterators and has
+    // no impact on point lookups.
+    fn table_filter(&self, props: &TableProperties) -> bool;
+}
+
+pub extern "C" fn table_filter(ctx: *mut c_void, props: *const DBTableProperties) -> c_int {
+    unsafe {
+        let filter = &*(ctx as *mut Box<Box<TableFilter>>);
```
Comment (huachaohuang) [inline src/table_filter.rs]: Can we just use one `Box` here?

**Comment by zhangjinpeng1987** [comment]
@huachaohuang PTAL

**Comment by zhangjinpeng1987** [comment]
PTAL @huachaohuang @UncP

**Comment by UncP** [comment]
LGTM

File: librocksdb_sys/crocksdb/c.cc [hunk 3]
```diff
@@ -2789,6 +2789,54 @@ void crocksdb_readoptions_set_ignore_range_deletions(
   opt->rep.ignore_range_deletions = v;
 }
 
+struct TableFilter {
```
Comment (huachaohuang) [inline librocksdb_sys/crocksdb/c.cc]: Can we create another struct like `TableFilterContext` to wrap the related context, then put a `shared_ptr` inside `TableFilter`?

**Comment by huachaohuang** [comment]
Ping @zhangjinpeng1987

**Comment by zhangjinpeng1987** [comment]
@huachaohuang PTAL

**Pull request merged**